src/main/scala/mesosphere/marathon/state/AppDefinition.scala

```diff
@@ -657,7 +657,9 @@ object AppDefinition extends GeneralPurposeCombinators {
 
   private val complyWithGpuRules: Validator[AppDefinition] = conditional[AppDefinition](_.gpus > 0) {
     isTrue[AppDefinition]("GPU resources only work with the Mesos containerizer") { app =>
-      app.container.exists{
+      // If 'container' is not defined, the Mesos containerizer is used.
+      // By using forall, this case will yield true.
+      app.container.forall{
```

Comment:
This depends on how we want to formulate the rule. I'd say, GPU is allowed for _everything but Docker_, so I'd write
```
app.container match {
  case Some(_: Docker) => false
  case _ => true
}
```

I we wanted to say _we specifically allow MesosDocker, MesosAppC, Mesos but nothing else_, then the proposed solution would be ok'ish (yet it's cumbersome to have the `forall` plus a pattern match for when the option is non empty)